### ISSUE TITLE
Next steps

### ISSUE THREAD

**Issue by ultimike** [communityinclusion/escoach#56]
@thinkcollege @escoach assuming that tasks #46, #51, #52, #53, and #54 are behind us, what are the next steps? Here's what I'm thinking:

1. Either @thinkcollege or I rebase the `rack` branch onto `devel` to get things up-to-date. I'm not sure if @thinkcollege has done this lately, but I know it's been 2-3 weeks since I have.
2. Should we focus on the charts or the trends stuff next? Or is there something else other than those two things.

I know that you wanted to have a beta by end of last week - is what we have now sufficient for that?

thanks,  
-mike

**Comment by thinkcollege** [comment]
Mike, you can do the rebase from rack into devel.  I haven't touched it, 
and you may be more aware of what is going on on the devel branch.  Let 
me know about any conflicts.

We can hold off on trends.  We would like to have a working search 
interface and table to demo in the next few weeks.  It might be better 
to hide the Trends selector until it is working as we want it to, but I 
will let @escoach address that. We also have to start thinking about 
user interface stuff--labeling and descriptions, UI design.  As a 
non-statistician I find the interface pretty brutal right now.

On 8/31/21 3:48 AM, Michael Anello wrote:
> EXTERNAL SENDER
>
> @thinkcollege 
> <https://nam10.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fthinkcollege&data=04%7C01%7Cpaul.foos%40umb.edu%7C8e78b8cd14cc46cb532208d96c53c181%7Cb97188711ee94425953c1ace1373eb38%7C0%7C0%7C637659929255308627%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=kAYdoXtqQKdO%2BluCzLTt4W7Bp8%2FjwyH9LF6ROwPxeL0%3D&reserved=0> 
> @escoach 
> <https://nam10.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fescoach&data=04%7C01%7Cpaul.foos%40umb.edu%7C8e78b8cd14cc46cb532208d96c53c181%7Cb97188711ee94425953c1ace1373eb38%7C0%7C0%7C637659929255308627%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=SimY4%2FDUJOQltoR6T4zwaM3hHYH1TuU0qPwS4S8WaGw%3D&reserved=0> 
> assuming that tasks #46 
> <https://nam10.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fcommunityinclusion%2Fescoach%2Fissues%2F46&data=04%7C01%7Cpaul.foos%40umb.edu%7C8e78b8cd14cc46cb532208d96c53c181%7Cb97188711ee94425953c1ace1373eb38%7C0%7C0%7C637659929255318585%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=btUyZZz51g%2F0bWEKqik3gtIygtETC1fsJt2Nh3eTnj4%3D&reserved=0>, 
> #51 
> <https://nam10.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fcommunityinclusion%2Fescoach%2Fissues%2F51&data=04%7C01%7Cpaul.foos%40umb.edu%7C8e78b8cd14cc46cb532208d96c53c181%7Cb97188711ee94425953c1ace1373eb38%7C0%7C0%7C637659929255318585%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=e%2F3adhH9hGaBsd7VPgcaUT3J6Aisweq5y0o%2F7wjST30%3D&reserved=0>, 
> #52 
> <https://nam10.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fcommunityinclusion%2Fescoach%2Fissues%2F52&data=04%7C01%7Cpaul.foos%40umb.edu%7C8e78b8cd14cc46cb532208d96c53c181%7Cb97188711ee94425953c1ace1373eb38%7C0%7C0%7C637659929255328541%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=zDQjEOP8zeJN%2FuSvtR5%2BPvGuVEH13upTBn77Xia11%2BA%3D&reserved=0>, 
> #53 
> <https://nam10.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fcommunityinclusion%2Fescoach%2Fissues%2F53&data=04%7C01%7Cpaul.foos%40umb.edu%7C8e78b8cd14cc46cb532208d96c53c181%7Cb97188711ee94425953c1ace1373eb38%7C0%7C0%7C637659929255328541%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=Pc9kY4bab%2FUZjBmZTDVbZQhAR68U1Jo5wU4r8Ei1czg%3D&reserved=0>, 
> and #54 
> <https://nam10.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fcommunityinclusion%2Fescoach%2Fissues%2F54&data=04%7C01%7Cpaul.foos%40umb.edu%7C8e78b8cd14cc46cb532208d96c53c181%7Cb97188711ee94425953c1ace1373eb38%7C0%7C0%7C637659929255328541%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=wMlWLlC5pDVaVRfMg%2F5TgnwXQFVRjujByeewDh7xU%2FQ%3D&reserved=0> 
> are behind us, what are the next steps? Here's what I'm thinking:
>
>  1. Either @thinkcollege
>     <https://nam10.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fthinkcollege&data=04%7C01%7Cpaul.foos%40umb.edu%7C8e78b8cd14cc46cb532208d96c53c181%7Cb97188711ee94425953c1ace1373eb38%7C0%7C0%7C637659929255338497%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=tUt6QY9FSdIcLBZyTFlkI9GTDGOQ48aNR%2FhRrxDhi%2Bw%3D&reserved=0>
>     or I rebase the |rack| branch onto |devel| to get things
>     up-to-date. I'm not sure if @thinkcollege
>     <https://nam10.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fthinkcollege&data=04%7C01%7Cpaul.foos%40umb.edu%7C8e78b8cd14cc46cb532208d96c53c181%7Cb97188711ee94425953c1ace1373eb38%7C0%7C0%7C637659929255338497%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=tUt6QY9FSdIcLBZyTFlkI9GTDGOQ48aNR%2FhRrxDhi%2Bw%3D&reserved=0>
>     has done this lately, but I know it's been 2-3 weeks since I have.
>  2. Should we focus on the charts or the trends stuff next? Or is
>     there something else other than those two things.
>
> I know that you wanted to have a beta by end of last week - is what we 
> have now sufficient for that?
>
> thanks,
> -mike
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub 
> <https://nam10.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fcommunityinclusion%2Fescoach%2Fissues%2F56&data=04%7C01%7Cpaul.foos%40umb.edu%7C8e78b8cd14cc46cb532208d96c53c181%7Cb97188711ee94425953c1ace1373eb38%7C0%7C0%7C637659929255338497%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=1SfeY4dRIEuaEiLDOGWoP7eosifx3O3K4DA9wsGNN0k%3D&reserved=0>, 
> or unsubscribe 
> <https://nam10.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAANZAVGMFK2CTBV7RUOISEDT7SCNVANCNFSM5DDQCJWQ&data=04%7C01%7Cpaul.foos%40umb.edu%7C8e78b8cd14cc46cb532208d96c53c181%7Cb97188711ee94425953c1ace1373eb38%7C0%7C0%7C637659929255348454%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=v9ChLnd3kQIFr8IxcJUyrQBg2ZqSVoa8yMDvlIgWIN4%3D&reserved=0>.
> Triage notifications on the go with GitHub Mobile for iOS 
> <https://nam10.safelinks.protection.outlook.com/?url=https%3A%2F%2Fapps.apple.com%2Fapp%2Fapple-store%2Fid1477376905%3Fct%3Dnotification-email%26mt%3D8%26pt%3D524675&data=04%7C01%7Cpaul.foos%40umb.edu%7C8e78b8cd14cc46cb532208d96c53c181%7Cb97188711ee94425953c1ace1373eb38%7C0%7C0%7C637659929255348454%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=s5ZCeuF0PE%2BL%2FSuGUohcelbnCvwySNdMJWaMJE4vGFQ%3D&reserved=0> 
> or Android 
> <https://nam10.safelinks.protection.outlook.com/?url=https%3A%2F%2Fplay.google.com%2Fstore%2Fapps%2Fdetails%3Fid%3Dcom.github.android%26referrer%3Dutm_campaign%253Dnotification-email%2526utm_medium%253Demail%2526utm_source%253Dgithub&data=04%7C01%7Cpaul.foos%40umb.edu%7C8e78b8cd14cc46cb532208d96c53c181%7Cb97188711ee94425953c1ace1373eb38%7C0%7C0%7C637659929255358411%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=q1dd2iqTr7Oc5OSym%2BrmNPxeyhbVRa%2FpM2aeE7d2Fy0%3D&reserved=0>. 
>
>
-- 
Paul Foos
Web Developer
Institute for Community Inclusion
University of Massachusetts Boston
100 Morrissey Boulevard
Boston, Massachusetts  02125
Office: (617) 287-4359
Cell phone: (508) 521-0719

**Comment by ultimike** [comment]
@thinkcollege @mgcobblestone I just rebased `rack` onto `devel` and force-pushed `devel` back to `origin`. I don't think that broke anything (at least on the Dashboard).

I'll leave it to @thinkcollege or @escoach to open new tasks for where you'd like us to focus our attention next. 

@thinkcollege I'm not clear what you mean by the "search interface and table". I believe that the queries are returning the correct data to the tables at this point, @escoach can confirm.

How about for the next two tasks (both simple), we do:

1. Hide the "time-frame" option.
2. Remove the "debug" stuff. 

-mike

**Comment by thinkcollege** [comment]
I meant "search interface and labels."  Not "interface and table." That was a typo.  I will work with @escoach to come up with suggestions on how to make the interface more friendly.  Alberto and I have a meeting scheduled for tomorrow.

**Comment by escoach** [comment]
@ultimike I asked a colleague to run a few more case scenarios to double-check that all data showing on the dashboard is good. Probably it's not needed. But I would feel better knowing that the dashboard passed the test of someone who has not been deeply involved in figuring out the equations like you and I. After @thinkcollege and I meet tomorrow Paul will let you know about priorities.

**Comment by thinkcollege** [comment]
Hi @ultimike,  @escoach and I talked today and agreed on next steps, as follows, roughly in order of priority:

1. Move the code, as is (or with basic modifications for usability) over to the live site so we can get some of our beta users to look at it with live data. If we could have it in place this week or early next week we would have some active test users to experience it before we roll it out. 
2. Bit of a backtrack, sorry: we want to re-include trend data, at least the monthly category.  We realized this is necessary, so development on this needs to go forward, and be usable by end of September or mid-October. 
3. We would like you to provide a time/cost estimate on building the visual charts.  We would like to be able to evaluate our options on whether to end the project with just functional table display without charts, or to go ahead with the chart display in addition to the table.  We have stayed within budget so far and would like to finish at or under budget.  A consideration that is unknown to me but we would need to know is if you have already gone ahead with work on the charts and if there are sunk costs already committed to building them.  Let us know if that is the case.
4. I created a Provider taxonomy term, "Observer" on the live site which we would like to assign to internal users and others whose survey data we don't want included in the dashboard.  Could you put in some code to exclude this Provider? 
5. Improve the user experience (e.g. dashboard labeling, instructions) This will be the last step because we want a functioning interface first, then take a look from a user perspective to get a better idea of how to help the user.

**Comment by escoach** [comment]
Hi @ultimike following up on @thinkcollege list of tasks, Paul suggested that I add a clarification regarding item 4. Specifically, rather than filtering out the "observer" data altogether, could you filter them out only in the "All" querries? 

Rationale: "observers" may include potential future users who want to try es-coach by using it and see how it works. By filtering out their data only in the "All" column, the "observers" would be able to see both their individual data and their team data and compare with everybody else data ("All" data) without actually compromising the "All" data (We are not sure that their data are completely clean and usable during their trial period). Please let me know if this is confusing.

**Comment by ultimike** [comment]
@escoach @thinkcollege Thanks for all this info - very helpful.

@mgcobblestone Can you provide an hour estimate for implementing the charting capability?

I created new tasks for:

- Moving the code to the live site: #57 
- Monthly trends: #58
- "Observer" provider: #59'

I didn't create a task for improving the user experience. I'm going to assume that one of you will create tasks as necessary for that.

Did I miss anything?

thanks,  
-mike

**Comment by ultimike** [comment]
@mgcobblestone Friendly nudge for the charting capability hour estimate 😄

**Comment by mgcobblestone** [comment]
@ultimike I don't think should take more than 6-8 hrs.

**Comment by ultimike** [comment]
@thinkcollege @escoach I think at this point, we need a prioritized list of tasks to work on, including where the charting fits in (see Martin's previous comment).

thanks,  
-mike

**Comment by thinkcollege** [comment]
Thanks for the updates @ultimike and @mgcobblestone We will try to get you a task priority list later today.

**Comment by thinkcollege** [comment]
Add some text and labeling to dashboard as shown in this document [Dashboard layout 091421.docx](https://github.com/communityinclusion/escoach/files/7165050/Dashboard.layout.091421.docx)
	
Get Monthly trend table working  But hold off until after Thursday 9/16 because there will be a demo on that day. Also clarification is needed on math for the monthly display.
	
Quarterly trend tables. However, pending how the monthly trends tables go in terms of complexity and time needs/resources, we might want to drop the quarterly trends.
	
Adding charts may be the last step, but we want this if possible.  Could we have a re-estimate of total budget with and without charts?

**Comment by escoach** [comment]
@ultimike note that the Dashboard layout that @thinkcollege shared includes also aligning the What, Who, and Where menus on top, below the welcome paragraph. This is different than in the current version where the Who and Where are on top and the What is below the Who. Not sure if this note is useful, but I was not sure if the word doc was clear enough. Thanks

**Comment by ultimike** [comment]
@thinkcollege @escoach I have made the following changes as requested in the Word docx and pushed to the dev site:

(@mgcobblestone tagging you just so you are aware of these changes.)

1. I hid the title "DashboardForm". 
2. I rearranged the layout.
2. I added the Welcome text as a block - this can be easily edited by normal Drupal methods.
3. I also added the Legend text as a block. Doing it this way, I am not able to easily confine it to the right-hand column, but it makes it much easier for you to edit the text via the UI, so I figured this was a better approach.
3. I renamed the form to just "Dashboard" (this is what appears in the browser title bar).
4. I've commented out the debugging stuff on the form.
5. I slightly extended the height of the "What" options box.

I have pushed the code changes related to all this to the Dev site, it is up to Paul to pull them into the live site.

I have created both the Welcome and Legend blocks on both the Dev and Live sites.

Regarding the next tasks you listed above:

1. Monthly trends - #58  - waiting on feedback/information from @escoach 
2. Quarterly trends - on hold pending #58 
3. Charting - Martin estimated 6-8 hours of work to get this working.

-mike

-mike

**Comment by thinkcollege** [comment]
Thanks @ultimike  In my last point I asked for a total job estimate with and without the charts, not just Martin's part of it.  @escoach may have a handle on this, but I am not sure where we will end up with total cost of the job.

**Comment by ultimike** [comment]
Understood - I just sent you an email with a summary.

-mike

**Comment by escoach** [comment]
@ultimike Thank you for the changes in the layout of the dashboard. Looks great. Also, I appreciate the option to edit the text. Very useful.

**Comment by thinkcollege** [comment]
@ultimike  Below is @escoach 's specification of what is needed for the monthly trends display.  Let us know if this adds significantly to hours or if there are difficulties with this.
----------------------------
 Here is what to ask Mike in regard to the trend data tables:
Generate a monthly trend data as visualized in Table 1 in this Google sheet: https://docs.google.com/spreadsheets/d/1xKtCi2jgxygtJyhGS4qTN7I6rqtiGHx-LpsuW7877lk/edit#gid=1970680850 (Tab "Trends0916", See cell A2:F17) Note:


I included the most recent 12 months' worth of data: from August 1, 2020 to July 31, 2021.
	
The time value for each month is an average of 3 months. For ex., the time values in October is the average of Aug, Sep, and Oct. 
	
If it is not too complicated, include the n selected activities per each month for the "me" data (Column F in the sheet.)
	
All formulas are available in the cells of the sheet. 
	
Remember to exclude the "Observers"
	
Once we agree on this format, I will be happy to generate a few more tables for other types (e.g. Who).

PS: The quarterly tables will be very easy to generate. for example, the data for the quarter Oct 2020-Dec 2020 is A10:F10.

**Comment by ultimike** [comment]
Moving the trends stuff to its proper place: #58 

If we're all set with the rest of the stuff in this task, please close @thinkcollege or @escoach 

-mike
